### PR TITLE
Discussion only: working build for all OS

### PR DESCRIPTION
**Pull request by ann0see**
I now got the code to a stage where all jobs run successfully. The problem was that the official qt download server was down. 

I also had to revert a few changes you did. 

The commit history is also not great ;-).

### CODE REVIEW

**Comment by nefarius2001** [comment]
nice, thanks.
I used "git merge --squash" to make it a single commit to understand whats happening (7d5201a1019452ba5100a414976aa70b092c64b0)

I like the included flatpak. One job that dois it all might be the way.

I am still not sure if the single build step is a good idea or not.
Spent my evening recovering the latest "known good" state where all 5 flavors build...

**Comment by nefarius2001** [comment]
Seems like I completely wasted my time and this is still a QT-download thing...
The same commit that once worked, does not work any more

https://github.com/nefarius2001/jamulus/actions/runs/493433720
https://github.com/nefarius2001/jamulus/actions/runs/499844222

**Comment by ann0see** [comment]
At least for me, last time I ran the action from my changes it worked. I did change the download server too, so I don’t think that’s the problem

**Comment by ann0see** [comment]
https://github.com/ann0see/jamulus/actions/runs/497175772

This seemed to work

File: windows/autorelease_windows.ps1 [hunk 1]
```diff
@@ -7,9 +7,9 @@ echo "Install Qt..."
 # Install Qt
 pip install aqtinstall
 echo "Get Qt 64 bit..."
-aqt install --outputdir C:\Qt 5.15.2 windows desktop win64_msvc2019_64
+aqt install --outputdir C:\Qt 5.15.2 windows desktop win64_msvc2019_64 -b https://mirrors.ocf.berkeley.edu/qt/
```
Comment (ann0see) [inline windows/autorelease_windows.ps1:10]: @nefarius2001 can you add this mirror in your fork too?

File: windows/autorelease_windows.ps1 [hunk 2]
```diff
@@ -7,9 +7,9 @@ echo "Install Qt..."
...
 echo "Get Qt 32 bit..."
-aqt install --outputdir C:\Qt 5.15.2 windows desktop win32_msvc2019
+aqt install --outputdir C:\Qt 5.15.2 windows desktop win32_msvc2019 -b https://mirrors.ocf.berkeley.edu/qt/
```
Comment (ann0see) [inline windows/autorelease_windows.ps1:12]: Same here

File: android/automated_build/install-qt.sh [hunk 3]
```diff
@@ -233,13 +233,13 @@ DOWNLOAD_DIR=`mktemp -d 2>/dev/null || mktemp -d -t 'install-qt'`
 function compute_url(){
     local COMPONENT=$1
     local CURL="curl -s -L"
-    local BASE_URL="http://download.qt.io/online/qtsdkrepository/${HOST_OS}/${TARGET_PLATFORM}"
+    local BASE_URL="https://mirrors.ocf.berkeley.edu/qt/online/qtsdkrepository/${HOST_OS}/${TARGET_PLATFORM}"
```
Comment (ann0see) [inline android/automated_build/install-qt.sh:236]: Not sure if we're allowed to change it here (not looked into the license yet). But can you change the mirror here?

**Comment by nefarius2001** [comment]
I still am confused and see if i can reproduce
https://github.com/ann0see/jamulus/actions/runs/500466428
https://github.com/nefarius2001/jamulus/actions/runs/501641654

Anyways, you have a working state, I haven't.
How about you 
- create a new branch based on either  nefarius2001/automated_build2 or corrados/master
- run "git merge --squash patch-2" => you have all changes in workspace
- stage & commit ( everything at once or in multiple commits up to you)

I don't mind if you stage a pull request to currados directly or back to me.
what do you think?

**Comment by ann0see** [comment]
> create a new branch based on corrados/master
I think that's the best way thing we can do. 

I'd open a new PR to your repo so you can open a PR and do a few changes.

**Comment by nefarius2001** [comment]
Weird, that the same commit worked on your repo, but not mine ( see links above)

**Comment by ann0see** [comment]
> Weird, that the same commit worked on your repo, but not mine ( see links above)

Argh. I think I know where the error lies now: you need to rename the flatpack branch from patch-2 to automated_build2

File: .github/workflows/autobuild.yml [hunk 4]
```diff
@@ -131,3 +123,34 @@ jobs:
...
+      with:
+        bundle: "io.jamulus.Jamulus.flatpak"
+        manifest-path: "distributions/io.jamulus.Jamulus.json"
+    - uses: actions/download-artifact@v2
+      with:
+        name: io.jamulus.Jamulus
+        path: ~/io.jamulus.Jamulus
+    - name: List
+      run: ls ~/; ls ~/io.*/*
+    - name:                       Upload Flatpack
```
Comment (ann0see) [inline .github/workflows/autobuild.yml:147]: @nefarius2001 

I forgot to add the upload of flatpak to my PR...

you should add it to master

**Comment by nefarius2001** [comment]
see my pull request to you ;)

**Pull request closed without merging**